### PR TITLE
Adopt a 100 column wide code style.

### PR DESCRIPTION
**Pull request by KineticTheory**
### Background

* Current developer have requested that we adopt a 100 column wide code format that is enforced.

### Description of changes

* Update top level `.clang-format` to enforce a 100 column wide style.
* Update Emacs defaults to use 100 columns.
* Manually format files in the ds++ directory to use the new style.
* Make a first cut at replacing all dividers with 100 col dividers.

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [x] Travis/Appveyor CI checks pass
  * [x] Code coverage does not decrease
  * [x] [Valgrind test passes](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Toss3 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Trinitite checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] Code reviewed/approved, sufficient DbC checks, testing, documentation
* WIP
  * [x] Do non-updated files create clang-format warnings -- no!
  * [x] Do we manually update all of the text blocks to be 100 cols?
  * [x] Do we need to update the VIM style sheet to be 100 cols?

### CODE REVIEW

**Comment by KineticTheory** [comment]
@alexrlongne Should I update the vim configs to support a 100 column guideline? (see WIP section above)
@keadyk Do you have any recommendation about manually updating comment blocks? (see WIP section above)

**Comment by alexrlongne** [comment]
@KineticTheory That'd be great! If should be a very simple change.

**Comment by KineticTheory** [comment]
> @KineticTheory That'd be great! If should be a very simple change.

Can you provide a diff for me?

**Comment by alexrlongne** [comment]
> > @KineticTheory That'd be great! If should be a very simple change.
> 
> Can you provide a diff for me? Does this look right?
> 
> ```diff
> autocmd BufWritePre * :call RemoveTrailing()
>  
>  " ----------------------------------------------------------------------
> -" sets a gray warning line at 81 columns
> +" sets a gray warning line at 101 columns
>  if exists('+colorcolumn')
>    highlight ColorColumn ctermbg=gray
> -  set colorcolumn=81
> +  set colorcolumn=101
>  else
> -  au BufWinEnter * let w:m2=matchadd('ErrorMsg', '\%>80v.\+', -1)
> +  au BufWinEnter * let w:m2=matchadd('ErrorMsg', '\%>100v.\+', -1)
>  endif
> ```

That's exactly what I did. I loaded that version and it successfully moved the column line.

File: config/print_target_properties.cmake [hunk 1]
```diff
@@ -108,6 +108,6 @@ function(print_targets_properties)
   endforeach()
 endfunction()
 
-#------------------------------------------------------------------------------#
+#--------------------------------------------------------------------------------------------------#
 # End config/print_target_properties.cmake
-#------------------------------------------------------------------------------#
\ No newline at end of file
+#--------------------------------------------------------------------------------------------------#
```
Comment (keadyk) [inline config/print_target_properties.cmake:113]: I'm not sure if this is important, but GitHub points out there is no newline here (and there wasn't before, either :D )

Comment (KineticTheory) [reply config/print_target_properties.cmake:113]: I'm going to ignore that warning for now.  There are probably a lot of these issues in our source code.

File: environment/elisp/draco-mode.el [hunk 2]
```diff
@@ -466,11 +466,11 @@ function or method. Width is set by draco-code-comment-width.
 "Function for inserting a class desicription boilerplate.  Width is set
```
Comment (keadyk) [inline environment/elisp/draco-mode.el:466]: I noticed that description is misspelled here, if you want to change it :)

Comment (KineticTheory) [reply environment/elisp/draco-mode.el:466]: I'll fix it. Let me know when you are done reviewing.  I'll submit all the changes in a single push.

**Comment by keadyk** [comment]
@KineticTheory This looks good to me! I'm glad clang format isn't going to warn on all the unchanged files in the interim!

**Comment by KineticTheory** [comment]
@keadyk I fixed the spelling error, the missing newline and I fixed a path for the rzansel bashrc.  You can merge when ready.

**Comment by keadyk** [comment]
Thanks, merging! :)

**Pull request merged**